File: app/src/main/java/de/bikebean/app/db/settings/settings/replace_if_newer_settings/Status.java
package de.bikebean.app.db.settings.settings.replace_if_newer_settings;

import androidx.annotation.NonNull;

import de.bikebean.app.db.settings.settings.ReplaceIfNewerSetting;
import de.bikebean.app.db.sms.Sms;
import de.bikebean.app.db.sms.SmsFactory;
import de.bikebean.app.db.state.State;
import de.bikebean.app.db.state.StateFactory;

public class Status extends ReplaceIfNewerSetting {

    private static final @NonNull
    State.KEY key = State.KEY._STATUS;

    public Status(final @NonNull Sms sms) {
        super(sms, StateFactory.createNumberState(sms, key, 0.0, State.STATUS.CONFIRMED));
    }

    public Status() {
        super(SmsFactory.createNullSms(),
                StateFactory.createNumberState(
                        SmsFactory.createNullSms(), key, 0.0, State.STATUS.UNSET
                )
        );
    }
}

File: app/src/main/java/de/bikebean/app/db/settings/settings/replace_if_newer_settings/Status.kt
package de.bikebean.app.db.settings.settings.replace_if_newer_settings

import de.bikebean.app.db.settings.settings.ReplaceIfNewerSetting
import de.bikebean.app.db.sms.Sms
import de.bikebean.app.db.state.State
import de.bikebean.app.db.state.State.KEY
import de.bikebean.app.db.state.StateFactory

class Status : ReplaceIfNewerSetting {

    constructor(sms: Sms) : super(
            StateFactory.createNumberState(sms, key, 0.0, State.STATUS.CONFIRMED)
    )

    constructor() : super(StateFactory.createUnsetState(key, 0.0))

    companion object {
        private val key = KEY._STATUS
    }
}